src/main/java/ca/corefacility/bioinformatics/irida/util/IridaFiles.java

```diff
@@ -0,0 +1,79 @@
+package ca.corefacility.bioinformatics.irida.util;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.util.List;
+
+import ca.corefacility.bioinformatics.irida.model.sequenceFile.SequenceFile;
+import ca.corefacility.bioinformatics.irida.model.sequenceFile.SequencingObject;
+import ca.corefacility.bioinformatics.irida.repositories.filesystem.IridaFileStorageUtility;
+
+/**
+ * Static class which has file object operations that require
+ * access to the iridaFileStorageUtility but in a static context
+ */
+
+public final class IridaFiles {
+
+	private static IridaFileStorageUtility iridaFileStorageUtility;
+
+	public static void setIridaFileStorageUtility(IridaFileStorageUtility iridaFileStorageUtility) {
+		IridaFiles.iridaFileStorageUtility = iridaFileStorageUtility;
+	}
+	private IridaFiles() {
+	}
+
+	/**
+	 * Gets the file size of the file from the iridaFileStorageUtility
+	 *
+	 * @param file The path to the file
+	 * @return file size as a human readable string
+	 */
+	public static String getFileSize(Path file) {
+		return iridaFileStorageUtility.getFileSize(file);
+	}
+
+	/**
+	 * Checks if the file is gzipped in iridaFileStorageUtility
+	 *
+	 * @param file The path to the file
+	 * @return if file is gzipped or not
+	 * @throws IOException if file cannot be read
+	 */
+	public static boolean isGzipped(Path file) throws IOException {
+		return iridaFileStorageUtility.isGzipped(file);
+	}
+
+	/**
+	 * Gets the file input stream from iridaFileStorageUtility
+	 *
+	 * @param file The path to the file
+	 * @return the file input stream
+	 */
+	public static InputStream getFileInputStream(Path file) {
+		return iridaFileStorageUtility.getFileInputStream(file);
+	}
+
+	/**
+	 * Gets the file extension from iridaFileStorageUtility
+	 *
+	 * @param files List of sequencingObjects to get file extensions for
+	 * @return the common extension of the files
+	 * @throws IOException if file(s) cannot be read
+	 */
+	public static String getFileExtension(List<? extends SequencingObject> files) throws IOException {
+		return iridaFileStorageUtility.getFileExtension(files);
+	}
+
+	/**
+	 * Appends a sequence file to another file
+	 *
+	 * @param target The path to the file into which to append
+	 * @param file The sequence file to append
+	 * @throws IOException if file cannot be read/written
+	 */
+	public static void appendToFile(Path target, SequenceFile file) throws IOException {
```

Comment:
Taking a look at this file it definitely simplifies some stuff. I think there's a couple methods in here that should maybe not be here though Specifically ones dealing with any of our model objects or ones that modify files. So appendToFile should maybe not be here. I know that would mean adding it back to the concatenator classes and such, but it might be better.

Possibly also with getFileExtension, but that one isn't as bad. I think that would be ok.

Reply:
Removed appendFiles from IridaFiles in 97e0014e1c5aa5584ad4b4f07bf67f3d54fbaf70 but left the getFileExtension in there since it's not modifying anything.